File: scripts/ThemesPreviewsProvider.js
function ThemesPreviewsProvider() { }

ThemesPreviewsProvider._path = '/dwr';

ThemesPreviewsProvider.getThemesPreviewsInfo = function(callback) {
    DWREngine._execute(ThemesPreviewsProvider._path, 'ThemesPreviewsProvider', 'getThemesPreviewsInfo', callback);
}

ThemesPreviewsProvider.getPagePreview = function(callback) {
    DWREngine._execute(ThemesPreviewsProvider._path, 'ThemesPreviewsProvider', 'getPagePreview', callback);
}

ThemesPreviewsProvider.getThemeStyleVariations = function(p0, callback) {
    DWREngine._execute(ThemesPreviewsProvider._path, 'ThemesPreviewsProvider', 'getThemeStyleVariations', p0, callback);
}

ThemesPreviewsProvider.changeTheme = function(p0, p1, p2, p3, p4, callback) {
    DWREngine._execute(ThemesPreviewsProvider._path, 'ThemesPreviewsProvider', 'changeTheme', p0, p1, p2, p3, p4, callback);
}

ThemesPreviewsProvider.saveTheme = function(p0, p1, callback) {
    DWREngine._execute(ThemesPreviewsProvider._path, 'ThemesPreviewsProvider', 'saveTheme', p0, p1, callback);
}

File: src/java/com/idega/content/themes/business/ThemesPreviewsProvider.java
package com.idega.content.themes.business;


import com.idega.business.IBOService;
import java.rmi.RemoteException;

public interface ThemesPreviewsProvider extends IBOService {
	/**
	 * @see com.idega.content.themes.business.ThemesPreviewsProviderBean#getThemesPreviewsInfo
	 */
	public String getThemesPreviewsInfo() throws RemoteException;

	/**
	 * @see com.idega.content.themes.business.ThemesPreviewsProviderBean#getThemeStyleVariations
	 */
	public String getThemeStyleVariations(String themeID) throws RemoteException;

	/**
	 * @see com.idega.content.themes.business.ThemesPreviewsProviderBean#changeTheme
	 */
	public String changeTheme(String themeID, String styleGroupName, String styleMember, String themeName, boolean radio, boolean checked) throws RemoteException;

	/**
	 * @see com.idega.content.themes.business.ThemesPreviewsProviderBean#saveTheme
	 */
	public boolean saveTheme(String themeID, String themeName) throws RemoteException;

	/**
	 * @see com.idega.content.themes.business.ThemesPreviewsProviderBean#setSelectedStyle
	 */
	public String setSelectedStyle(String themeID, boolean applyToPage) throws RemoteException;
}